projects/ids-enterprise-ng/src/lib/button/soho-button.d.ts

```diff
@@ -5,26 +5,97 @@
  * interface of the Soho button control.
  */
 
+type SohoButtonOptionsStyle = 'default' | 'btn' | 'btn-primary' | 'btn-secondary' | 'btn-tertiary' | 'btn-destructive' | null | undefined;
+
+type SohoButtonOptionsType = 'default' | 'btn-icon' | 'btn-menu' | 'btn-actions' | 'btn-toggle' | 'icon-favorite' | 'btn-editor' | 'input' | null | undefined;
+
 /**
  * Button Options
  */
 interface SohoButtonOptions {
+  /** An optional identifier for the button. */
+  id?: string;
+
+  /** The icon to use for on state toggle buttons. */
   toggleOnIcon?: string;
+
+  /** The icon to use for off state on toggle buttons. */
   toggleOffIcon?: string;
+
+  /** If true the selection will be used to replace the content. */
   replaceText?: boolean;
 
-  
   /** Hides menu arrow usually displayed to the right of a menu button/icon. */
   hideMenuArrow?: boolean;
+
+  /** button style: defaults to "default". */
+  style?: SohoButtonOptionsStyle;
+
+  /** button type: defaults to "default". */
+  type?: SohoButtonOptionsType,
+
+  /** Ripple */
+  ripple?: boolean,
+
+  /** validate? */
+  validate?: boolean
+
+  /** Audible? */
+  audible?: boolean;
+
+  /** Displayed text. */
+  text?: string;
+
+  /** Associated icon. */
+  icon?: string;
+
+  /** Disabled? */
+  disabled?: boolean;
 }
 
 /**
  * This interface represents the public API exposed by the
  * button.
  */
-interface SohoButtonStatic {
+declare class SohoButtonStatic {
+  /** Internal settings. */
   settings: SohoButtonOptions;
+
+  /** Associated HTML element. */
+  element: HTMLButtonElement;
+
+  /** Destroys any resources held by the button. */
   destroy(): void;
+
+  /** Returns true if the button is disabled; otherwise false. */
+  get disabled(); boolean;
```

Comment:
semi-colon after ()

Reply:
fixed - not sure why lint doesn't pick these up.